lib/active_merchant/billing/gateways/merchant_one.rb

```diff
@@ -42,14 +41,45 @@ def purchase(money, creditcard, options = {})
         add_address(post, creditcard, options)
         add_customer_data(post, options)
         add_amount(post, money, options)
-        commit('sale', money, post)
+        commit('sale', post)
       end
 
       def capture(money, authorization, options = {})
         post = {}
         post.merge!(:transactionid => authorization)
         add_amount(post, money, options)
-        commit('capture', money, post)
+        commit('capture', post)
+      end
+
+      def void(authorization, option = {})
+        post = {}
+        add_transaction_data(post, authorization)
+        commit('void', post)
+      end
+
+      def refund(money, authorization, options = {})
+        post = {}
+        post.merge!(:transactionid => authorization)
+        add_amount(post, money, options)
+        commit('refund', post)
+      end
+
+      def store(creditcard, options = {})
+        post = {}
+        type = nil
+        post['customer_vault'] = 'add_customer'
```

Comment:
convert this to a function like `add_customer_vault` to follow the patterh of the other functions